python-avd/pyavd/_eos_designs/structured_config/network_services/router_ospf.py

```diff
@@ -20,85 +23,110 @@ class RouterOspfMixin(Protocol):
                 if not process_id:
                     msg = f"Missing or invalid 'ospf.process_id' or 'vrf_id' under vrf '{vrf.name}"
                     raise AristaAvdInvalidInputsError(msg)
+                process._update(id=process_id, passive_interface_default=True)
 
-                process = {
-                    "id": process_id,
-                    "vrf": vrf.name if vrf.name != "default" else None,
-                    "passive_interface_default": True,
-                    "router_id": self.get_vrf_router_id(vrf, vrf.ospf.router_id, tenant.name),
-                    "no_passive_interfaces": ospf_interfaces,
-                    "bfd_enable": vrf.ospf.bfd or None,  # Historic behavior is to only output if True.
-                    "max_lsa": vrf.ospf.max_lsa,
-                }
-
-                process_redistribute = {}
-
-                if vrf.ospf.redistribute_bgp.enabled:
-                    process_redistribute["bgp"] = {"enabled": True}
-                    if route_map := vrf.ospf.redistribute_bgp.route_map:
-                        process_redistribute["bgp"]["route_map"] = route_map
-
-                if vrf.ospf.redistribute_connected.enabled:
-                    process_redistribute["connected"] = {"enabled": True}
-                    if route_map := vrf.ospf.redistribute_connected.route_map:
-                        process_redistribute["connected"]["route_map"] = route_map
-
-                process["redistribute"] = process_redistribute or None
-
-                # Strip None values from process before adding to list
-                process = {key: value for key, value in process.items() if value is not None}
-
-                append_if_not_duplicate(
-                    list_of_dicts=ospf_processes,
-                    primary_key="id",
-                    new_dict=process,
-                    context="OSPF Processes defined under network services",
-                    context_keys=["id"],
-                )
+                self._set_ospf_vrf(process, vrf, tenant.name)
+                self._set_ospf_redistribute(process, vrf)
 
+                self.structured_config.router_ospf.process_ids.append(process)
         # If we have static_routes in default VRF and not EPVN, and underlay is OSPF
         # Then add redistribute static to the underlay OSPF process.
         if self._vrf_default_ipv4_static_routes["redistribute_in_underlay"] and self.shared_utils.underlay_routing_protocol in ["ospf", "ospf-ldp"]:
-            ospf_processes.append({"id": self.inputs.underlay_ospf_process_id, "redistribute": {"static": {"enabled": True}}})
-        if ospf_processes:
-            return {"process_ids": ospf_processes}
+            process = EosCliConfigGen.RouterOspf.ProcessIdsItem(id=self.inputs.underlay_ospf_process_id)
+            process.redistribute.static.enabled = True
+            self.structured_config.router_ospf.process_ids.append(process)
+
+    def _set_ospf_redistribute(
```

Comment:
this should be called `update_ ... ` since you are updating the given `process` object instead of `set`ting directly on self.structured_config.

Reply:
same for the others.

Reply:
Fixed!